medusa/providers/torrent/html/cinemaz.py

```diff
@@ -0,0 +1,184 @@
+            'login': urljoin(self.url, 'auth/login'),
+            'search': urljoin(self.url, 'torrents'),
+        }
+
+        # Proper Strings
+        self.proper_strings = ['PROPER', 'REPACK', 'REAL', 'RERIP']
+
+        # Miscellaneous Options
+        self.freeleech = False
+
+        # Torrent Stats
+
+        # Cache
+        self.cache = tv.Cache(self)  # only poll CinemaZ every 10 minutes max
+
+    def search(self, search_strings, age=0, ep_obj=None, **kwargs):
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+        if not self.login():
+            return results
+
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            for search_string in search_strings[mode]:
+                if mode != 'RSS':
+                    log.debug('Search string: {search}',
+                              {'search': search_string})
+
+                search_params = {
+                    'in': 1,
+                    'search': search_string,
+                    'type': 2,
+                    'discount[]': 1 if self.freeleech else None,
+                    'tv_type[]': {'episode': 1, 'season': 2}.get(mode.lower())
+                }
+
+                if not search_string:
+                    del search_params['search']
+
+                response = self.session.get(self.urls['search'], params=search_params)
+                if not response or not response.text:
+                    log.debug('No data returned from provider')
+                    continue
+
+                results += self.parse(response.text, mode)
+
+        return results
+
+    def parse(self, data, mode):
+        """
+        Parse search results for items.
+
+        :param data: The raw response from a search
+        :param mode: The current mode used to search, e.g. RSS
+
+        :return: A list of items found
+        """
+        items = []
+
+        with BS4Parser(data, 'html5lib') as html:
+            if not html:
+                log.debug('No html data parsed from provider')
+                return items
+
+            torrents = html('tr')
+            if not torrents or len(torrents) < 2:
+                log.debug('Data returned from provider does not contain any torrents')
+                return items
+
+            # Skip column headers
+            for row in torrents[1:]:
+                # Skip extraneous rows at the end
+                if len(row.contents) < 10:
+                    continue
+
+                try:
+                    title = row.find(class_='torrent-filename').text.strip()
+                    download_url = row.find(class_='torrent-download-icon').get('href')
+                    size = convert_size(row.contents[11].text.strip(), -1)
```

Comment:
Please use get_text(strip=True) instead, use `default=-1` and move it after leechers.